datafusion/core/src/datasource/physical_plan/parquet/statistics.rs

```diff
@@ -600,6 +600,18 @@ make_data_page_stats_iterator!(
     Index::DOUBLE,
     f64
 );
+make_data_page_stats_iterator!(
+    MinByteArrayDataPageStatsIterator,
+    |x: &PageIndex<ByteArray>| { x.min.clone() },
```

Comment:
I spent some time digging in the code and I found that this `clone` is cloning a `Bytes` (aka it is ref counted, and is not actually copying the bytes around)